### PR TITLE
10321 jax-rs cdi managedobject 2

### PR DESCRIPTION
**Pull request by benjamin-confino**

### CODE REVIEW

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2015, 2016 IBM Corporation and others.
+ * Copyright (c) 2015, 2019 IBM Corporation and others.
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java]: its 2020 now!

File: dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java [hunk 2]
```diff
@@ -398,8 +402,12 @@ public static ClassLoader getAndSetLoader(ClassLoader newCL) {
      * @return true if it is a proxy
      */
     public static boolean isWeldProxy(Class<?> clazz) {
-        boolean result = clazz.getSimpleName().contains(PROXY_CLASS_SIGNATURE);
-        return result;
+        for ( Class implementedInterface : clazz.getInterfaces() ) {
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java]: This should just be `clazz.instanceOf(ProxyObject.class)`

File: dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java [hunk 3]
```diff
@@ -398,8 +402,12 @@ public static ClassLoader getAndSetLoader(ClassLoader newCL) {
...
+            if ( implementedInterface.getName().equals( WELD_PROXY_INTERFACE_NAME ) ) {
+                return true;
+            }
+        }
+        return false;
```
Comment (Azquelt) [inline dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java]: This should just be `return clazz.instanceOf(ProxyObject.class);`

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

File: dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java [hunk 4]
```diff
@@ -52,13 +52,17 @@
...
+import org.jboss.weld.bean.proxy.ProxyObject;
+
 /**
  * Common constants and utility methods
  */
 public class CDIUtils {
 
     private static final TraceComponent tc = Tr.register(CDIUtils.class);
 
+    private static final String WELD_PROXY_INTERFACE_NAME = ProxyObject.class.getName();
```
Comment (tevans78) [inline dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/CDIUtils.java]: What is this constant for?

File: dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java [hunk 5]
```diff
@@ -273,6 +271,34 @@ private Object getClassFromCDI(Class<?> clazz) {
         return serviceObject;
     }
 
+    /**
+     * @param clazz
+     * @return
+     */
+    @FFDCIgnore(value = { Exception.class })
+    private <T> ManagedObject<T> getClassFromServiceObject(Class<T> clazz, Object serviceObject) {
```
Comment (tevans78) [inline dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java]: How about `createManagedObjectForServiceObject` or just `createManagedObject`?

File: dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java [hunk 6]
```diff
@@ -273,6 +271,34 @@ private Object getClassFromCDI(Class<?> clazz) {
...
+
+        if (! clazz.equals(serviceObject.getClass())) {
```
Comment (tevans78) [inline dev/com.ibm.ws.jaxrs.2.0.cdi/src/com/ibm/ws/jaxrs20/cdi/component/JaxRsFactoryImplicitBeanCDICustomizer.java]: Are you sure that is a safe check?

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#build

**Comment by benjamin-confino** [comment]
#run-libby-bot

**Pull request merged**